litestar/_signature/models/base.py

```diff
@@ -69,31 +71,25 @@ def _build_error_message(cls, keys: Sequence[str], exc_msg: str, connection: ASG
             An ErrorMessage
         """
 
-        message: ErrorMessage = {"message": exc_msg}
-
-        if len(keys) > 1:
-            key_start = 0
-
-            if keys[0] == "data":
-                key_start = 1
-                message["source"] = "body"
-
-            message["key"] = ".".join(keys[key_start:])
-        elif keys:
-            key = keys[0]
-            message["key"] = key
-
-            if key in connection.query_params:
-                message["source"] = cast("Literal['cookie', 'body', 'header', 'query']", "query")
-
-            elif key in cls.fields and isinstance(cls.fields[key].kwarg_model, ParameterKwarg):
-                if cast(ParameterKwarg, cls.fields[key].kwarg_model).cookie:
-                    source = "cookie"
-                elif cast(ParameterKwarg, cls.fields[key].kwarg_model).header:
-                    source = "header"
-                else:
-                    source = "query"
-                message["source"] = cast("Literal['cookie', 'body', 'header', 'query']", source)
+        message: ErrorMessage = {"message": exc_msg.split(" - ")[0]}
+
+        if not keys:
+            return message
+
+        message["key"] = key = ".".join(keys)
+
+        if key in connection.query_params:
+            message["source"] = "query"
+        elif key in cls.fields and isinstance(cls.fields[key].kwarg_model, ParameterKwarg):
+            kwarg_model = cast("ParameterKwarg", cls.fields[key].kwarg_model)
+            if kwarg_model.cookie:
+                message["source"] = "cookie"
+            elif kwarg_model.header:
+                message["source"] = "header"
+            else:
+                message["source"] = "query"
+        else:
+            message["source"] = "body"
```

Comment:
Refactor of this method has removed the need to have the key start with `data` to know that the source should be `"body"`.